compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -372,10 +375,12 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
             // AbstractValidatingLambdaMetafactory.validateMetafactoryArgs
 
             val DesugaredSelect(prefix, _) = fun: @unchecked
-            genLoad(prefix)
+            val prefixTK = genLoad(prefix)
+            stack.push(prefixTK)
```

Comment:
This was missing before, although it's a bit theoretical, since the arguments to `Closure`s are always identifiers in practice, which means they don't contain any `Return` nor `Try` that would need an accurate `stack`.